extensions/WindowsSandboxExtension/Providers/WindowsSandboxComputeSystem.cs

```diff
@@ -0,0 +1,204 @@
+
+            return properties.AsEnumerable();
+        }).AsAsyncOperation();
+    }
+
+    public IAsyncOperation<ComputeSystemStateResult> GetStateAsync()
+    {
+        return Task.Run(() =>
+        {
+            return new ComputeSystemStateResult(State);
+        }).AsAsyncOperation();
+    }
+
+    public IAsyncOperation<ComputeSystemOperationResult> ConnectAsync(string options)
+    {
+        return Task.Run(() =>
+        {
+            try
+            {
+                lock (_windowsSandboxStartLock)
+                {
+                    // Windows Sandbox is not running.
+                    if (_windowsSandboxExeProcess == null || _windowsSandboxExeProcess.HasExited)
+                    {
+                        var system32Path = Environment.GetFolderPath(Environment.SpecialFolder.System);
+                        var windowsSandboxExePath = Path.Combine(system32Path, Constants.WindowsSandboxExe);
+
+                        _windowsSandboxExeProcess = new();
+                        _windowsSandboxExeProcess.StartInfo.FileName = windowsSandboxExePath;
+                        _windowsSandboxExeProcess.EnableRaisingEvents = true;
+                        _windowsSandboxExeProcess.Exited += WindowsSandboxProcessExited;
+
+                        Thread.Sleep((int)TimeSpan.FromSeconds(30).TotalMilliseconds);
+
+                        State = ComputeSystemState.Running;
+                        TraceLogging.StartingWindowsSandbox();
+
+                        _windowsSandboxExeProcess.Start();
+
+                        PInvoke.SetForegroundWindow((HWND)_windowsSandboxExeProcess.MainWindowHandle);
+                    }
+
+                    BringWindowsSandboxClientToForeground();
+
+                    return new ComputeSystemOperationResult();
+                }
+            }
+            catch (Exception ex)
+            {
+                _log.Error(ex, "Failed to start Windows Sandbox");
+                TraceLogging.ExceptionThrown(ex);
+                return new ComputeSystemOperationResult(ex, Resources.GetResource("WindowsSandboxFailedToStart", _log), "Failed to start Windows Sandbox");
+            }
+        }).AsAsyncOperation();
+    }
+
+    private void WindowsSandboxProcessExited(object? sender, EventArgs e)
+    {
+        State = ComputeSystemState.Stopped;
+        _windowsSandboxExeProcess?.Dispose();
+        _windowsSandboxExeProcess = null;
+    }
+
+    private Process? WindowsSandboxClientProcess()
+    {
+        return Process.GetProcessesByName("WindowsSandboxClient").FirstOrDefault();
+    }
+
+    private void BringWindowsSandboxClientToForeground()
+    {
+        var clientProcess = WindowsSandboxClientProcess();
+        var windowHandle = clientProcess?.MainWindowHandle ?? IntPtr.Zero;
+
+        PInvoke.SetForegroundWindow((HWND)windowHandle);
+    }
+
+    public IAsyncOperation<ComputeSystemOperationResult> TerminateAsync(string options)
+    {
+        return Task.Run(() =>
```

Comment:
For all your SDK methods you'd probably want to add a try/catch  and then return the *Result with the appropriate failure error message. I mention the reason why below.

Here is an example:
https://github.com/microsoft/devhome/blob/2a36c1aefa27e2bb6a33b145d0e5d6876f10c12d/HyperVExtension/src/HyperVExtension/Providers/HyperVProvider.cs#L66